tests/unit/pages/embeds/Plio.spec.js

```diff
@@ -318,4 +319,96 @@ describe("Plio.vue", () => {
     expect(checkAndSetPlayerAspectRatio).toHaveBeenCalled();
     expect(wrapper.vm.isAspectRatioChecked).toBeTruthy();
   });
+
+  it("calls updateEvent method for consecutive 'watching' events for a given session", async () => {
+    const plioId = "123";
+
+    // mock setPlayerAspectRatio method
+    jest
+      .spyOn(Plio.methods, "setPlayerAspectRatio")
+      .mockImplementation(() => jest.fn());
+
+    // mock createSession method
+    jest.spyOn(Plio.methods, "createSession").mockImplementation(() => {
+      return new Promise((resolve) => resolve());
+    });
+
+    // mock updateEvent method
+    let updateEvent = jest
+      .spyOn(Plio.methods, "updateEvent")
+      .mockImplementation(() => {
+        return;
+      });
+
+    // mock user service
+    jest
+      .spyOn(UserAPIService, "getUserByAccessToken")
+      .mockImplementation(() => {
+        return new Promise((resolve) => {
+          resolve({ data: dummyUser });
+        });
+      });
+
+    // set user
+    await store.dispatch("auth/setAccessToken", dummyAccessToken);
+
+    mount(Plio, {
+      shallow: true,
+      props: {
+        plioId: plioId,
+      },
+      data() {
+        return {
+          watchingEventDBId: 1,
+          sessionDBId: 1,
+          videoId: "_hJEyDOn6Ho",
+        };
+      },
+    });
+    await flushPromises();
+
+    // resolve the `GET` request waiting in the queue (for receiving plio details)
+    // using the fake response data
+    mockAxios.mockResponse(clonedeep(dummyPublishedPlio), mockAxios.queue()[0]);
+
+    // wait until the DOM updates after promises resolve
+    await flushPromises();
+    // assert that updateEvent method has been called with the correct params
+    expect(updateEvent).toHaveBeenCalledWith("watching", 1);
+  });
+
+  it("makes a PATCH call to update a watching event", () => {
+    // spy on the updateEvent method inside EventAPIService
+    let updateEventAPICall = jest.spyOn(EventAPIService, "updateEvent");
+
+    // we're testing the individual method using 'call'
+    // refer this - https://lmiller1990.github.io/vue-testing-handbook/v3/computed-properties.html#testing-with-call
+    // build a local context object - localThis. It will serve as 'this' in our test
+    let localThis = {
+      hasSessionStarted: true,
+      isAuthenticated: true,
+      previewMode: false,
+      sessionDBId: 3,
+      player: {
+        currentTime: 10,
+      },
+    };
+
+    let eventType = "watching";
+    let eventDBId = 1;
+
+    // invoke the method using the local context object and appropriate params
+    Plio.methods.updateEvent.call(localThis, eventType, eventDBId);
+
+    // the updateEvent method inside EventAPIService would've been called
+    // A patch request would've been made with the correct payload
```

Comment:
we don't need to explain which request types to expect in the comments - we have not done that for any API request tests + it is fairly obvious from the code + it ties the comment to the implementation (which we'll have to remember to update if we update the method)